pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -283,13 +283,17 @@ ContextImpl setupContext() throws PulsarClientException {
         Logger instanceLog = LoggerFactory.getILoggerFactory().getLogger(
                 "function-" + instanceConfig.getFunctionDetails().getName());
         Thread currentThread = Thread.currentThread();
+        ClassLoader clsLoader = currentThread.getContextClassLoader();
         Consumer<Throwable> fatalHandler = throwable -> {
             this.deathException = throwable;
             currentThread.interrupt();
         };
-        return new ContextImpl(instanceConfig, instanceLog, client, secretsProvider,
-                collectorRegistry, metricsLabels, this.componentType, this.stats, stateManager,
-                pulsarAdmin, clientBuilder, fatalHandler);
+        Thread.currentThread().setContextClassLoader(functionClassLoader);
+        ContextImpl contextImpl = new ContextImpl(instanceConfig, instanceLog, client, secretsProvider,
+            collectorRegistry, metricsLabels, this.componentType, this.stats, stateManager,
+            pulsarAdmin, clientBuilder, fatalHandler);
+        Thread.currentThread().setContextClassLoader(clsLoader);
+        return contextImpl;
```

Comment:
use a try-finally block so that the original context classloader is restored in the finally block. this type of solution is used in other locations where the context classloader is modified.

Reply:
thanks, addressed in the latest commit